kong/plugins/http-log/schema.lua

```diff
@@ -16,6 +17,15 @@ return {
           { retry_count = { type = "integer", default = 10 }, },
           { queue_size = { type = "integer", default = 1 }, },
           { flush_timeout = { type = "number", default = 2 }, },
-    }, }, },
+          { header_name = typedefs.header_name({ default = "Authorization"}), },
+          { header_value = { type = "string" }, },
+        },
+        entity_checks = {
+          -- header_name is required when header value is given
+          { conditional = {
+            if_field = "header_value", if_match = { ne = null },
+            then_field = "header_name", then_match = { required = true },
+          }, }, },
```

Comment:
Should we check that when `header_name` is given, `header_value` is also present? Otherwise user may mis configure the plugin without knowing and it is a silent failure.

Reply:
how would a user miss configure it if he does not set a header value? If he doesn't set either he'd get the same silent failure...

Reply:
But it seems that header name and value should always be set in pairs, so the fact that we are checking only one way makes little sense. Are we saying it is more likely for the user to forget setting header names but not the other way around?

Reply:
because the headername defaults to "Authorization", hence no check required

Reply:
and if you provide a value, then the name becomes required